### PR TITLE
feature/SG-220_refactor_pascal_detection_dataset_interface

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

File: src/super_gradients/recipes/dataset_params/pascal_voc_detection_dataset_params.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  input_dim: [320, 320]
+  cache: False
+  cache_dir:
+  transforms:
+  - DetectionPaddedRescale:
+      input_dim: ${dataset_params.train_dataset_params.input_dim}
+  - DetectionTargetsFormatTransform:
+      max_targets: 50
+      output_format:
+        _target_: super_gradients.training.utils.detection_utils.DetectionTargetsFormat # targets format
```
Comment (ofrimasad) [inline src/super_gradients/recipes/dataset_params/pascal_voc_detection_dataset_params.yaml:12]: don't we have a factory for this?

File: src/super_gradients/training/dataloaders/dataloader_factory.py [hunk 2]
```diff
@@ -412,3 +416,21 @@ def supervisely_persons_val(dataset_params: Dict = {}, dataloader_params: Dict =
                            train=False,
                            dataset_params=dataset_params,
                            dataloader_params=dataloader_params)
+
+
+def pascal_voc_train(dataset_params: Dict = {}, dataloader_params: Dict = {}):
```
Comment (ofrimasad) [inline src/super_gradients/training/dataloaders/dataloader_factory.py:421]: maybe call it pascal_voc_detection_train and pascal_voc_detection_val?

File: src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py [hunk 3]
```diff
@@ -18,13 +22,16 @@
...
+    def __init__(self, images_sub_directory: str, download: bool = False, *args, **kwargs):
         """Dataset for Pascal VOC object detection
 
         :param images_sub_directory:    Sub directory of data_dir that includes images.
         """
+
         self.images_sub_directory = images_sub_directory
         self.img_and_target_path_list = None
+        if download:
+            PascalVOCDetectionDataset.download(kwargs.get("data_dir"))
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py:34]: assert "data_dir" in kwargs

File: src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py [hunk 4]
```diff
@@ -132,3 +140,29 @@ def get_coord(box_coord):
                 new_label_path = (dest_labels_path / img_path.name).with_suffix('.txt')
                 img_path.rename(new_img_path)  # Move image to dest folder
                 _parse_and_save_labels(data_path, new_label_path, year, id)
+
+
+class PascalVOCUnifiedDetectionTrainDataset(ConcatDataset):
```
Comment (ofrimasad) [inline src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py:145]: where is the code of the previous version (before using yaml). I expected to see many lines being deleted

**Review by ofrimasad — COMMENTED** [review]
minor comments

File: src/super_gradients/training/dataloaders/dataloader_factory.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/dataloaders/dataloader_factory.py]: sure, renamed in https://github.com/Deci-AI/super-gradients/pull/332/commits/3cb87ded44227afef53deb1815d0ed8f2daf9f8f

File: src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py (hunk 3 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py]: added in https://github.com/Deci-AI/super-gradients/pull/332/commits/3cb87ded44227afef53deb1815d0ed8f2daf9f8f

File: src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py (hunk 4 above)
Comment (shaydeci) [reply src/super_gradients/training/datasets/detection_datasets/pascal_voc_detection.py:148]: This is because the main has already been done in https://github.com/Deci-AI/super-gradients/pull/278 .

I simply added the basic functionality of unifying the datasets (most common way to train on voc) as unified trainset (this was hard coded in the dataset interface).

File: src/super_gradients/recipes/dataset_params/pascal_voc_detection_dataset_params.yaml (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/recipes/dataset_params/pascal_voc_detection_dataset_params.yaml:12]: No, notice we resave here args for transforms.

I think the topic of resolving args for classes resolved by factories is a different topic, a one we should discuss how to do outside this PR as it might lead us into trouble when done wrong.

**Review by ofrimasad — APPROVED** [review]
LGTM
